### ISSUE TITLE
Ouch face is never shown due to directional damage indicator priority

### ISSUE THREAD

**Issue by BCG2000** [fabiangreffrath/crispy-doom#371]
When receiving more than 20 points of damage, the ouch face is never shown because the HUD gives priority to showing where the damage came from (face turning left and right, as well as facing toward the damage). This means your ouch face fix doesn't work in practice (https://github.com/fabiangreffrath/crispy-doom/issues/127).

Demonstration of the bug in MAP08 of DOOM2.WAD (I turn around after being hit to show the bug more clearly): https://www.youtube.com/watch?v=uNoWlpT__ko

**Comment by JNechaevsky** [comment]
@fabiangreffrath, try to add `priority = 8;` in this condition, this should fix ouch face: 
https://github.com/fabiangreffrath/crispy-doom/blob/master/src/doom/st_stuff.c#L1271

**Comment by BCG2000** [comment]
@fabiangreffrath, same here. But try doing what I did in MAP08 of DOOM2.WAD, you won't see the ouch face. So the bug only happens when the damage source is not yourself.

**Comment by JNechaevsky** [comment]
Ah, yes, that's because self-damage calls for a different code, here:
https://github.com/fabiangreffrath/crispy-doom/blob/master/src/doom/st_stuff.c#L1319

Hmph. Something seems to be a little bit different in Crispy code, need to look closer...

**Comment by JNechaevsky** [comment]
Figured out, you have to remove/comment out previous `7` priority. Try this:

```
    if (priority < 8)
    {
	if (plyr->damagecount
	    && plyr->attacker
	    && plyr->attacker != plyr->mo)
	{
	    // being attacked
---	    priority = 7;
	    
	    // [crispy] show "Ouch Face" as intended
	    if (st_oldhealth - plyr->health > ST_MUCHPAIN)
	    {
		st_facecount = ST_TURNCOUNT;
		faceindex = ST_OUCHOFFSET;
+++		priority = 8;
	    }
	    else
	    {
```

**Comment by fabiangreffrath** [comment]
`priority = 7` has to remain for the directional faces, but raising `priority = 8` for the ouch face seems to fix the issue. Thanks!

**Comment by JNechaevsky** [comment]
Acording to the [changelog](https://www.chocolate-doom.org/wiki/index.php/Crispy_Doom/Changelog), face hysteresis was fixed in 4.0. Just tried version 3.5 ([link](http://www.greffrath.com/~fabian/crispy-doom_3.5.zip)), and results are interesting:

- By taking self-damage, ouch face is appearing normally.
- By taking damage by the monster, ouch face is appearing only for few (one?) tics.

So the answes seems to be both "yes and no". :)

**Comment by fabiangreffrath** [comment]
Indeed, the ouch face is shown for only one tic:

When the player is hurt by more than 20 damage points, the condition `(st_oldhealth - plyr->health > ST_MUCHPAIN)` evaluates `true` and the ouch face is shown. Then, in the next tic, `st_oldhealth = plyr->health` in `ST_Ticker()` and since `priority = 7` the same code patch is entered again. But this time the condition evaluated `false` and the directional faces are shown.

Once you have seen this, it is quite obvious. :astonished:
